src/sys/unix.rs

```diff
@@ -845,6 +862,10 @@ fn sendmsg(
     // Safety: Same as above about `*const` -> `*mut`.
     msg.msg_iov = bufs.as_ptr() as *mut _;
     msg.msg_iovlen = min(bufs.len(), IovLen::MAX as usize) as IovLen;
+    // Safety: Same as above about `*const` -> `*mut`.
+    msg.msg_control = control_data.as_ptr() as *mut _;
+    msg.msg_controllen = control_data.len();
```

Comment:
At least on macOS this is a `u32`.
```suggestion
    msg.msg_controllen = control_data.len() as _;
```